### PR TITLE
Move AFK Check to server

### PR DESCRIPTION
**Pull request by BerkieBb**
**Describe Pull request**
This will remove the KickForAFK event which is very exploitable (but you can only really kick yourself with it, but unresponsible people could've used the event and defined source or something, it's just not good practice) and moved all checks for AFK to the server.

This PR has to be merged along with https://github.com/Qbox-project/qb-core/pull/21 and https://github.com/Qbox-project/qb-core/pull/22

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [yes] (Be honest)
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

### CODE REVIEW

**Review by Manason — APPROVED** [review]

File: server/afk.lua [hunk 1]
```diff
@@ -1,9 +1,83 @@
...
+RegisterNetEvent('QBCore:Server:OnPlayerLoaded', function()
+    loggedInPlayers[source] = true
+    updateCheckUser(source)
 end)
 
-QBCore.Functions.CreateCallback('qb-afkkick:server:GetPermissions', function(source, cb)
-    cb(QBCore.Functions.GetPermission(source))
+AddEventHandler('QBCore:Server:OnPlayerUnload', function(source)
+    loggedInPlayers[source] = false
+    checkUser[source] = false
```
Comment (D4isDAVID) [inline server/afk.lua:37]:
```suggestion
    checkUser[source] = nil
```
How about making this nil for players that aren't logged in? I think it makes more sense like this.

File: server/afk.lua [hunk 2]
```diff
@@ -1,9 +1,83 @@
...
+end)
+
+AddEventHandler('QBCore:Server:OnPermissionUpdate', function(source)
+    updateCheckUser(source)
+end)
+
+CreateThread(function()
+    for _, v in pairs(GetPlayers()) do
+        loggedInPlayers[v] = Player(v).state.isLoggedIn
```
Comment (D4isDAVID) [inline server/afk.lua:47]:
```suggestion
        loggedInPlayers[v] = Player(v).state.isLoggedIn
        if loggedInPlayers[v] then
            updateCheckPlayer(v)
        end
```
Should probably use `updateCheckPlayer()` here to ensure that players with the correct permissions are not checked when starting the resource after the players are already logged in.

Comment (D4isDAVID) [inline server/afk.lua:47]:
```suggestion
        v = tonumber(v)
        loggedInPlayers[v] = Player(v).state.isLoggedIn
```
Here you use `v` but later on you state that `GetPlayers()` returns string IDs and use `tonumber(v)` instead. I'm guessing it should be `tonumber(v)` here as well?

**Review by D4isDAVID — CHANGES_REQUESTED** [review]

File: server/afk.lua (hunk 1 above)
Comment (BerkieBb) [reply server/afk.lua:37]: No, then it will turn to true in the loop

Comment (D4isDAVID) [reply server/afk.lua:37]: True...

File: server/afk.lua [hunk 3]
```diff
@@ -1,9 +1,87 @@
...
+AddEventHandler('QBCore:Server:OnPermissionUpdate', function(source)
+    updateCheckUser(source)
+end)
+
+CreateThread(function()
+    for _, v in pairs(GetPlayers()) do
+        v = tonumber(v)
+        loggedInPlayers[v] = Player(v).state.isLoggedIn
+        if loggedInPlayers[v] then
+            updateCheckPlayer(v)
```
Comment (D4isDAVID) [inline server/afk.lua]:
```suggestion
            updateCheckUser(v)
```
Whoops, should be `updateCheckUser`, not `updateCheckPlayer`.
Player does indeed sound better though

**Review by D4isDAVID — CHANGES_REQUESTED** [review]

**Pull request merged**